File: ECommerce/Services/LoggingService.cs
using System.Text;
using System.Text.Json;
using E_Commerce.Interfaces;

namespace E_Commerce.Services
{
    public class LoggingService : ILoggingService
    {
        private readonly HttpClient _httpClient;

        public LoggingService(HttpClient httpClient)
        {
            _httpClient = httpClient;
        }

        public async Task LogAsync(string message)
        {
            var logEntry = new
            {
                logId = Guid.NewGuid().ToString(),
                message = "[ECOMMERCE]" + message
            };

            var content = new StringContent(JsonSerializer.Serialize(logEntry), Encoding.UTF8, "application/json");

            try
            {
                var response = await _httpClient.PostAsync("http://log-monitoring:5255/api/Logs", content);
                response.EnsureSuccessStatusCode();
                Console.WriteLine("Log enviado com sucesso.");
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine($"Erro ao enviar log: {ex.Message}");
            }
        }
    }
}

File: AUTH/back-end/App/LogService/LoggingService.py
import requests
import json
import uuid

class LoggingService:
    def __init__(self):
        self.log_service_url = "http://log-monitoring:5255/api/Logs"

    def log(self, message):
        log_entry = {
            "logId": str(uuid.uuid4()),
            "message": message
        }

        headers = {'Content-Type': 'application/json'}
        try:
            response = requests.post(self.log_service_url, data=json.dumps(log_entry), headers=headers)
            response.raise_for_status()
            print("Log enviado com sucesso.")
        except requests.exceptions.RequestException as e:
            print(f"Erro ao enviar log: {e}")

# Exemplo de uso
if __name__ == "__main__":
    logging_service = LoggingService("http://log-monitoring:5255/api/Logs")
    logging_service.log("Usuário admin tentou logar.")